tests/cmdline/commands/test_code.py

```diff
@@ -276,3 +231,52 @@ def setUp(self):
     def test_code_list_no_codes_error_message(self):
         result = self.cli_runner.invoke(code_list)
         self.assertEqual(1, result.output.count('# No codes found matching the specified criteria.'))
+
+
+@pytest.mark.parametrize('non_interactive_editor', ('sleep 1; vim -cwq',), indirect=True)
+def test_interactive_remote(clear_database_before_test, aiida_localhost, non_interactive_editor):
+    """Test interactive remote code setup."""
+    label = 'interactive_remote'
+    user_input = '\n'.join([label, 'description', 'arithmetic.add', 'yes', aiida_localhost.name, '/remote/abs/path'])
+    result = CliRunner().invoke(setup_code, input=user_input)
+    assert result.exception is None
+    assert isinstance(orm.Code.get_from_string('{}@{}'.format(label, aiida_localhost.name)), orm.Code)
+
+
+@pytest.mark.parametrize('non_interactive_editor', ('sleep 1; vim -cwq',), indirect=True)
+def test_interactive_upload(clear_database_before_test, aiida_localhost, non_interactive_editor):
+    """Test interactive code setup."""
+    label = 'interactive_upload'
+    dirname = os.path.dirname(__file__)
+    basename = os.path.basename(__file__)
+    user_input = '\n'.join([label, 'description', 'arithmetic.add', 'no', dirname, basename])
+    result = CliRunner().invoke(setup_code, input=user_input)
+    assert result.exception is None
+    assert isinstance(orm.Code.get_from_string('{}'.format(label)), orm.Code)
+
+
+@pytest.mark.parametrize('non_interactive_editor', ('sleep 1; vim -cwq',), indirect=True)
+def test_mixed(clear_database_before_test, aiida_localhost, non_interactive_editor):
+    """Test mixed (interactive/from config) code setup."""
+    from aiida.orm import Code
+    label = 'mixed_remote'
+    options = ['--description=description', '--on-computer', '--remote-abs-path=/remote/abs/path']
+    user_input = '\n'.join([label, 'arithmetic.add', aiida_localhost.name])
+    result = CliRunner().invoke(setup_code, options, input=user_input)
+    assert result.exception is None
+    assert isinstance(Code.get_from_string('{}@{}'.format(label, aiida_localhost.name)), Code)
+
+
+@pytest.mark.parametrize('non_interactive_editor', ('sleep 1; vim -cwq',), indirect=True)
+def test_code_duplicate_interactive(clear_database_before_test, aiida_local_code_factory, non_interactive_editor):
+    """Test code duplication interactive."""
+    label = 'code_duplicate_interactive_teor'
+    user_input = label + '\n\n\n\n\n\n'
+    code = aiida_local_code_factory('arithmetic.add', '/remote/abs/path', label='code', check_executable=False)
```

Comment:
e.g. here, you could point to an existing executable, say `python` or `verdi` or ...